Classes/PlusOnes.lua

```diff
@@ -2,126 +2,970 @@
+    if (not GL:strStartsWith(message, "!plusone")
+        and not GL:strStartsWith(message, "!PLUSONE")
+        and not GL:strStartsWith(message, "!po")
+        and not GL:strStartsWith(message, "!PO")
+        and not GL:strStartsWith(message, "!+1")
+    ) then
+        return;
+    end
+
+    local args = GL:strSplit(message, " ");
+
+    -- See if name is given.
+    if (#args > 1) then
+        local name = self:normalizedName(args[2]);
+        local plusOne = self:getPlusOnes(name);
+        GL:sendChatMessage(
+            string.format("Player %s's %s total is %d", GL:capitalize(name), GL.Settings:get("PlusOnes.identifier", "PO"), plusOne),
+            "WHISPER", nil, sender
+        );
+        return;
+    end
+
+    local name = sender;
+    if (not GL.isEra) then
+        name = GL:stripRealm(name);
+    end
+
+    name = self:normalizedName(name);
+    local plusOne = self:getPlusOnes(name);
+    GL:sendChatMessage(
+        string.format("Your %s total is %d", GL.Settings:get("PlusOnes.identifier", "PO"), plusOne),
+        "WHISPER", nil, sender
+    );
+end
+
+--- Materialize the plus one data to make it more accessible during runtime
+---
+---@return void
+function PlusOnes:materializeData()
+    GL:debug("PlusOnes:materializeData");
+
+    local DetailsByPlayerName = {}; -- Details including aliases by player name
+
+    --- Create entries from Totals data
+    for name, plusOne in pairs(DB:get("PlusOnes.Totals", {})) do
+        name = GL:normalizedName(name or "");
+        plusOne = self:toPlusOne(plusOne or 0);
+
+        if (type(name) == "string"
+            and not GL:empty(name)
+        ) then
+            if (not DetailsByPlayerName[name]) then
+                GL:tableSet(DetailsByPlayerName, name .. ".total", plusOne);
+            end
+        end
+    end
 
-    playerName = GL:normalizedName(playerName);
+    self.MaterializedData.DetailsByPlayerName = DetailsByPlayerName;
+end
 
-    if (not DB.PlusOnes[playerName]) then
-        DB.PlusOnes[playerName] = 0;
+--- Format a plus one.
+---
+---@param plusOne any
+---@return number if valid, else nil
+function PlusOnes:toPlusOne(plusOne)
+    plusOne = tonumber(plusOne);
+
+    if not plusOne then
+        return nil;
     end
 
-    DB.PlusOnes[playerName] = 0;
+    plusOne = math.max(0, math.floor(plusOne));
+    return plusOne;
+end;
+
+--- Clear all plus one data
+---
+---@return void
+function PlusOnes:clearPlusOnes()
+    DB.PlusOnes = {
+        Totals = {},
+        MetaData = {},
+    };
+    self.MaterializedData = {
+        DetailsByPlayerName = {},
+    };
+    
+    ---added this code below to broadcast when the ML clears all the plus one data.
+    if (GL.Settings:get("PlusOnes.automaticallyShareData")
+    and self:userIsAllowedToBroadcast()) then
+        self:broadcast(); 
+    end
+    ---
+
     self:triggerChangeEvent();
+end
+
+--[[--- Determine if a player is present in the table --- not sure if this is even called elsewhere
```

Comment:
Temporary?

Reply:
Deleted the whole section.  I confirmed the function was not called.